src/store/RootStore/QueryParamsStore/QueryParamsStore.ts

```diff
@@ -0,0 +1,31 @@
+import { parseQueryString } from "@utils/parseQueryString";
+import { Params } from "@utils/parseQueryString";
+import { makeAutoObservable } from "mobx";
+export default class QueryParamsStore {
+  _search = "";
+  _currency = "";
+  _params: Params = {
```

Comment:
если поля класса не `private`, не надо их называть с нижним подчеркиванием в начале
```suggestion
  search = "";
  currency = "";
  params: Params = {
```